modules/lang-mustache/src/main/java/org/elasticsearch/script/mustache/RestMultiSearchTemplateAction.java

```diff
@@ -65,6 +66,8 @@ public String getName() {
     @Override
     public RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
         MultiSearchTemplateRequest multiRequest = parseRequest(request, allowExplicitIndex);
+        // TODO I think we need to move this to the transport actions execute because we don't resolve scripts that early
+        CCSVersionCheckHelper.checkCCSVersionCompatibility(request, multiRequest);
```

Comment:
I see what you mean. These multi requests are tricky in general because they get converted to other requests and those are the ones that get sent over the wire. I ended up wondering when is the MultiSearchTemplateRequest ever serialized over the wire now that transport client has been removed? I agree, either we move that conversion logic to the rest layer, not even sure if possible or we move the check to the transport action.